lib/logstash/bundler.rb

```diff
@@ -0,0 +1,125 @@
+require "bundler"
+require "bundler/cli"
+
+module Bundler
+  # Patch bundler to write a .lock file specific to the version of ruby.
+  # This keeps MRI/JRuby/RBX from conflicting over the Gemfile.lock updates
+  module SharedHelpers
+    def default_lockfile
+      ruby = "#{LogStash::Environment.ruby_engine}-#{LogStash::Environment.ruby_abi_version}"
+      Pathname.new("#{default_gemfile}.#{ruby}.lock")
+    end
+  end
+
+  # Add the Bundler.reset! method which has been added in master but is not in 1.7.9.
+  class << self
+    unless self.method_defined?("reset!")
+      def reset!
+        @definition = nil
+      end
+    end
+
+    def reset_settings
+      # reset!
+      # clear_gemspec_cache
+
+      # remove_instance_variable(:@setup) if defined?(@setup)
+      # remove_instance_variable(:@locked_gems) if defined?(@locked_gems)
+      # remove_instance_variable(:@load) if defined?(@load)
+      remove_instance_variable(:@settings) if defined?(@settings)
+    end
+  end
+end
+
+module LogStash
+  module Bundler
+    # capture any $stdout from the passed block. also trap any exception in that block, in which case the trapped exception will be returned
+    # @param [Proc] the code block to execute
+    # @return [String, Exception] the captured $stdout string and any trapped exception or nil if none
+    def self.capture_stdout
+      begin
```

Comment:
no need for a top-level `begin` here since it is implied in all method definitions. This allows us to remove a level of indentation and would look like:

```ruby
def self.capture_output
  old_stdout = ...
  ...
ensure 
  $stdout = old_stdout
end
```